tests/phpunit/test-wc-stripe-payment-gateway.php

```diff
@@ -104,4 +104,157 @@ public function test_error_get_payment_intent_from_order() {
+			return 'USD';
+		};
+
+		add_filter( 'wc_get_base_location', $location_callback );
+		add_filter( 'get_woocommerce_currency', $currency_callback );
+
+		$payment_icons = $this->gateway->get_icon();
+
+		$this->assertContains( 'visa.svg', $payment_icons );
+		$this->assertContains( 'mastercard.svg', $payment_icons );
+		$this->assertContains( 'diners.svg', $payment_icons );
+		$this->assertContains( 'discover.svg', $payment_icons );
+		$this->assertContains( 'jcb.svg', $payment_icons );
+		$this->assertNotContains( 'amex.svg', $payment_icons );
+	}
+	
+	
+	public function test_get_icon_with_non_dollar_currency() {
+		$location_callback = function() {
+			return [
+				'country' => 'US',
+				'state'   => 'CA'
+			];
+		};
+
+		$currency_callback = function() {
+			return 'GBP';
+		};
+
+		add_filter( 'wc_get_base_location', $location_callback );
+		add_filter( 'get_woocommerce_currency', $currency_callback );
+
+		$payment_icons = $this->gateway->get_icon();
+
+		$this->assertContains( 'visa.svg', $payment_icons );
+		$this->assertContains( 'mastercard.svg', $payment_icons );
+		$this->assertContains( 'amex.svg', $payment_icons );
+		$this->assertNotContains( 'diners.svg', $payment_icons );
+		$this->assertNotContains( 'discover.svg', $payment_icons );
+		$this->assertNotContains( 'jcb.svg', $payment_icons );
+	}
+
+	public function test_get_icon_with_ca_store() {
+		$location_callback = function() {
+			return [
+				'country' => 'CA',
+				'state'   => ''
+			];
+		};
+
+		$currency_callback = function() {
+			return 'USD';
+		};
+
+		add_filter( 'wc_get_base_location', $location_callback );
+		add_filter( 'get_woocommerce_currency', $currency_callback );
+
+		$payment_icons = $this->gateway->get_icon();
+
+		$this->assertContains( 'visa.svg', $payment_icons );
+		$this->assertContains( 'amex.svg', $payment_icons );
+		$this->assertContains( 'mastercard.svg', $payment_icons );
+		$this->assertContains( 'diners.svg', $payment_icons );
+		$this->assertContains( 'discover.svg', $payment_icons );
+		$this->assertContains( 'jcb.svg', $payment_icons );
+	}
+
+	public function test_get_icon_with_jp_non_jpy_store() {
+		$location_callback = function() {
+			return [
+				'country' => 'JP',
+				'state'   => ''
+			];
+		};
+
+		$currency_callback = function() {
+			return 'JPY';
+		};
+
+		add_filter( 'wc_get_base_location', $location_callback );
+		add_filter( 'get_woocommerce_currency', $currency_callback );
+
+		$payment_icons = $this->gateway->get_icon();
+
+		// JCP only accepts JPY in Japanese stores:
+		$this->assertContains( 'visa.svg', $payment_icons );
+		$this->assertContains( 'amex.svg', $payment_icons );
+		$this->assertContains( 'mastercard.svg', $payment_icons );
+		$this->assertNotContains( 'diners.svg', $payment_icons );
+		$this->assertNotContains( 'discover.svg', $payment_icons );
+		$this->assertContains( 'jcb.svg', $payment_icons );
+	}
+
+	public function test_get_icon_with_jp_non_jpy_store() {
```

Comment:
IDK if the tests were ready for review. This function is already declared above.

Reply:
I hadn't run them yet haha, but good catch!